### PR TITLE
Remove singleton user

### PR DESCRIPTION
**Pull request by russelltankaimin**
## Objectives

1. Remove singleton usage of User.
2. Ensure GUI supports new User implementation
3. Test stub for User is also adapted for this new User implementation

### RFC

### CODE REVIEW

File: src/main/java/seedu/address/logic/Logic.java [hunk 1]
```diff
@@ -47,4 +48,9 @@ public interface Logic {
      * Set the user prefs' GUI settings.
      */
     void setGuiSettings(GuiSettings guiSettings);
+
+    /**
+     * Get user information
+     */
+    User getUser();
```
Comment (mynameizzhafeez) [inline src/main/java/seedu/address/logic/Logic.java:55]: Nice catch, I forgot that this also needed to be changed.

**Review by mynameizzhafeez — COMMENTED** [review]
LGTM. What else needs to be done?

**Review by mynameizzhafeez — APPROVED** [review]
LGTM.

**Comment by russelltankaimin** [comment]
> LGTM. What else needs to be done?

Nothing else. Left @zichen-3974 to approve because I modified her portion also since she used the ```getUser``` in her portion.

**Review by zichen-3974 — APPROVED** [review]
LGTM too.

**Pull request merged**